File: backend/app/schemas/analytics.py
from pydantic import BaseModel

class EnergyIntelligenceSummary(BaseModel):
    average_consumption_kwh: float | None
    latest_consumption_kwh: float | None
    consumption_change_percent: float | None

    average_generation_kwh: float | None
    latest_generation_kwh: float | None
    generation_change_percent: float | None

    consumption_trend: str
    generation_trend: str

    generation_anomaly: bool

File: frontend/src/types/analytics.ts
export interface EnergyIntelligenceSummary {
  average_consumption_kwh: number | null;
  latest_consumption_kwh: number | null;
  consumption_change_percent: number | null;

  average_generation_kwh: number | null;
  latest_generation_kwh: number | null;
  generation_change_percent: number | null;

  consumption_trend: string;
  generation_trend: string;

  generation_anomaly: boolean;
}